Dokan/Dokan/Scenes/ProductDetails/TitleQuantity/ProductTitleQuantityView.swift

```diff
@@ -18,18 +19,63 @@ class ProductTitleQuantityView: UIView {
     @IBOutlet private(set) weak var productStockBackgroundView: UIView!
     @IBOutlet private(set) weak var productStockCount: UILabel!
 
+    // MARK: Properities
+
+    lazy var productViewModel: ProductDetailsViewModel = .init()
+
     // MARK: Init
 
     override init(frame: CGRect) {
         super.init(frame: frame)
-
-        stylingProductDetails()
+        commonInit()
     }
 
     required init?(coder: NSCoder) {
         super.init(coder: coder)
+        commonInit()
+    }
+
+    private func commonInit() {
+        Bundle.main.loadNibNamed("ProductTitleQuantityView", owner: self, options: nil)
+        addSubview(titleContentView)
+        titleContentView.frame = bounds
+        titleContentView.autoresizingMask = [.flexibleHeight, .flexibleWidth]
 
         stylingProductDetails()
+        initViewModel()
+    }
+
+    func initView() {
+        let model: tQViewModel = productViewModel.getproductViewModel()
+        configureView(viewModel: model)
+        changeStockBackGroundColor(stockCount: model.stockCount)
+    }
+
+    func initViewModel() {
+
+        productViewModel.uploadLoadingState = { [weak self] () in
+            guard let self = self else { return }
+
+            DispatchQueue.main.async { [weak self] in
+                guard let self = self else { return }
+
+                switch self.productViewModel.state {
+
+                case .loading, .error, .empty:
+                    self.startSkeletonView()
+                case .populated:
+                    self.stopSkeletonView()
+                }
+            }
+        }
+
+        productViewModel.reloadViewClosure = { [weak self] () in
+            DispatchQueue.main.async { [weak self] in
+                self?.initView()
+            }
+        }
+
+        productViewModel.initFetchSingleProduct()
```

Comment:
This code needs more enhancement to handle the case when the view model gets updated and follow single responsibility.